packages/twenty-front/src/modules/favorites/hooks/useFavorites.ts

```diff
@@ -1,215 +1,165 @@
-import { mapFavorites } from '@/favorites/utils/mapFavorites';
+import { useGetObjectRecordIdentifierByNameSingular } from '@/object-metadata/hooks/useGetObjectRecordIdentifierByNameSingular';
 import { useObjectMetadataItem } from '@/object-metadata/hooks/useObjectMetadataItem';
-import { useObjectNameSingularFromPlural } from '@/object-metadata/hooks/useObjectNameSingularFromPlural';
-import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { isStandardObject } from '@/object-metadata/utils/isStandardObject';
+import { useCreateOneRecord } from '@/object-record/hooks/useCreateOneRecord';
+import { useDeleteOneRecord } from '@/object-record/hooks/useDeleteOneRecord';
 import { useFindManyRecords } from '@/object-record/hooks/useFindManyRecords';
-import { PaginatedRecordTypeResults } from '@/object-record/types/PaginatedRecordTypeResults';
-import { isDeeplyEqual } from '~/utils/isDeeplyEqual';
+import { useUpdateOneRecord } from '@/object-record/hooks/useUpdateOneRecord';
+import { FieldMetadataType } from '~/generated-metadata/graphql';
+import { isDefined } from '~/utils/isDefined';
 
-import { favoritesState } from '../states/favoritesState';
-
-export const useFavorites = ({
-  objectNamePlural,
-}: {
-  objectNamePlural: string;
-}) => {
+export const useFavorites = () => {
   const currentWorkspaceMember = useRecoilValue(currentWorkspaceMemberState);
 
-  const [favorites, setFavorites] = useRecoilState(favoritesState);
+  const favoriteObjectNameSingular = 'favorite';
+
+  const { objectMetadataItem: favoriteObjectMetadataItem } =
+    useObjectMetadataItem({
+      objectNameSingular: favoriteObjectNameSingular,
+    });
 
-  const {
-    updateOneRecordMutation: updateOneFavoriteMutation,
-    createOneRecordMutation: createOneFavoriteMutation,
-    deleteOneRecordMutation: deleteOneFavoriteMutation,
-  } = useObjectMetadataItem({
-    objectNameSingular: CoreObjectNameSingular.Favorite,
+  const { deleteOneRecord } = useDeleteOneRecord({
+    objectNameSingular: favoriteObjectNameSingular,
   });
 
-  const { triggerOptimisticEffects } = useOptimisticEffect({
-    objectNameSingular: CoreObjectNameSingular.Favorite,
+  const { updateOneRecord: updateOneFavorite } = useUpdateOneRecord({
+    objectNameSingular: favoriteObjectNameSingular,
   });
-  const { performOptimisticEvict } = useOptimisticEvict();
 
-  const { objectNameSingular } = useObjectNameSingularFromPlural({
-    objectNamePlural,
+  const { createOneRecord: createOneFavorite } = useCreateOneRecord({
+    objectNameSingular: favoriteObjectNameSingular,
   });
 
-  const { objectMetadataItem: favoriteTargetObjectMetadataItem } =
-    useObjectMetadataItem({
-      objectNameSingular,
-    });
+  const { records: favorites } = useFindManyRecords<Favorite>({
+    objectNameSingular: favoriteObjectNameSingular,
+  });
 
-  const apolloClient = useApolloClient();
+  const favoriteRelationFields = useMemo(
```

Comment:
I am not a big fan of the naming
FavoriteRelationFieldMetadataItem? (let's remove the plural and do a first instead of a filter) (it should log an error (but not throw if we have multiple results)